utility/gov.go

```diff
@@ -10,1798 +12,475 @@ import (
-			return types.ErrUpdateParam(err)
-		}
-		return nil
-	case types.MessagePauseValidatorFee:
-		i, ok := value.(*wrapperspb.StringValue)
-		if !ok {
-			return types.ErrInvalidParamValue(value, i)
-		}
-		err := store.SetMessagePauseValidatorFee(i.Value)
-		if err != nil {
-			return types.ErrUpdateParam(err)
-		}
-		return nil
-	case types.MessageUnpauseValidatorFee:
-		i, ok := value.(*wrapperspb.StringValue)
-		if !ok {
-			return types.ErrInvalidParamValue(value, i)
-		}
-		err := store.SetMessageUnpauseValidatorFee(i.Value)
-		if err != nil {
-			return types.ErrUpdateParam(err)
-		}
-		return nil
-	case types.MessageStakeServiceNodeFee:
-		i, ok := value.(*wrapperspb.StringValue)
-		if !ok {
-			return types.ErrInvalidParamValue(value, i)
-		}
-		err := store.SetMessageStakeServiceNodeFee(i.Value)
-		if err != nil {
-			return types.ErrUpdateParam(err)
-		}
-		return nil
-	case types.MessageEditStakeServiceNodeFee:
-		i, ok := value.(*wrapperspb.StringValue)
-		if !ok {
-			return types.ErrInvalidParamValue(value, i)
-		}
-		err := store.SetMessageEditStakeServiceNodeFee(i.Value)
-		if err != nil {
-			return types.ErrUpdateParam(err)
-		}
-		return nil
-	case types.MessageUnstakeServiceNodeFee:
-		i, ok := value.(*wrapperspb.StringValue)
-		if !ok {
-			return types.ErrInvalidParamValue(value, i)
-		}
-		err := store.SetMessageUnstakeServiceNodeFee(i.Value)
-		if err != nil {
-			return types.ErrUpdateParam(err)
-		}
-		return nil
-	case types.MessagePauseServiceNodeFee:
-		i, ok := value.(*wrapperspb.StringValue)
-		if !ok {
-			return types.ErrInvalidParamValue(value, i)
-		}
-		err := store.SetMessagePauseServiceNodeFee(i.Value)
-		if err != nil {
-			return types.ErrUpdateParam(err)
-		}
-		return nil
-	case types.MessageUnpauseServiceNodeFee:
-		i, ok := value.(*wrapperspb.StringValue)
-		if !ok {
-			return types.ErrInvalidParamValue(value, i)
-		}
-		err := store.SetMessageUnpauseServiceNodeFee(i.Value)
-		if err != nil {
-			return types.ErrUpdateParam(err)
-		}
-		return nil
-	case types.MessageChangeParameterFee:
-		i, ok := value.(*wrapperspb.StringValue)
-		if !ok {
-			return types.ErrInvalidParamValue(value, i)
-		}
-		err := store.SetMessageChangeParameterFee(i.Value)
-		if err != nil {
-			return types.ErrUpdateParam(err)
-		}
-		return nil
-	case types.MessageDoubleSignFeeOwner:
-		owner, ok := value.(*wrapperspb.BytesValue)
-		if !ok {
-			return types.ErrInvalidParamValue(value, owner)
-		}
-		err := store.SetMessageDoubleSignFeeOwner(owner.Value)
-		if err != nil {
-			return types.ErrUpdateParam(err)
-		}
-		return nil
-	default:
-		return types.ErrUnknownParam(paramName)
+
+	i, ok := value.(*wrapperspb.Int32Value)
+	if ok {
+		return types.ErrUpdateParam(store.SetParam(types.BlocksPerSessionParamName, (int(i.Value))))
+	}
+	s, ok := value.(*wrapperspb.StringValue)
+	if ok {
+		return types.ErrUpdateParam(store.SetParam(types.BlocksPerSessionParamName, s.Value))
 	}
+	b, ok := value.(*wrapperspb.BytesValue)
```

Comment:
NIT: 

If/else here, but switch statement in other places. of code:

Would prefer to keep it consistent

Reply:
Nice catch, I agree!

Also, you made me look at this code long enough to realize that I was always returning `types.ErrUpdateParam(store.SetParam(types.BlocksPerSessionParamName, t.Value))` and even if `store.SetParam(types.BlocksPerSessionParamName, t.Value)` returned nil, `UpdateParam` would have returned an error regardless. Currently not caught in tests since they are failing due to planned obsolescence (related to pre_persistence).